src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -922,11 +931,14 @@ public void AssemblyLoading_MultipleVersions_WorseMatchInSameDirectory(bool shad
                 StringBuilder sb = new StringBuilder();
 
                 var tempDir = temp.CreateDirectory();
-                var epsilonFile = tempDir.CreateFile("Epsilon.dll").CopyContentFrom(testFixture.Epsilon.Path);
-                var delta1File = tempDir.CreateFile("Delta.dll").CopyContentFrom(testFixture.Delta1.Path);
+                var tempDir1 = tempDir.CreateDirectory("a");
+                var tempDir2 = tempDir.CreateDirectory("b");
+                var epsilonFile = tempDir1.CreateFile("Epsilon.dll").CopyContentFrom(testFixture.Epsilon.Path);
+                var delta1File = tempDir1.CreateFile("Delta.dll").CopyContentFrom(testFixture.Delta1.Path);
+                var delta2File = tempDir2.CreateFile("Delta.dll").CopyContentFrom(testFixture.Delta2.Path);
```

Comment:
Previously this test was non-deterministic because the order of searched DLLs depended on generated directory names. This changes it to use specific directory names that have a consistent order.

Reply:
There isn't a need to do it in this PR, but it would be nice if we could dig a pit of success here--maybe by reducing direct usage of 'TempDirectory', and including some helper for putting files on disk into AssemblyLoadTestFixture or some other wrapper type, which will push us toward patterns that give deterministic results.

Reply:
I'm open to ideas on how to make this smarter. I have a follow up PR where I'm working on `AssemblyLoadTestFixture` that will fix some issues and I'm happy to incorporate determinism enhancements with it. 

It's tricky though ... everything within `AssemblyLoadTestFixture` is deterministic. That's actually in good shape. The problem is as soon as you create a new directory it's non-deterministic relative to `AssemblyLoadTestFixture`. Putting state into the type is a bit dicey because it's re-used between tests. 

Maybe need to introduce an intermediate type. One that is created off of an `AssemblyLoadTestFixture` instance per test that can hold state and create determenistic structures. Part of doing that is going to be hiding all the `TempDirectory` APIs in that type. I need to do that for other reasons.